simplified-downloader-core/src/main/java/org/nypl/simplified/downloader/core/DownloaderHTTP.java

```diff
@@ -223,7 +223,7 @@ private static String getContentType(
             this.log.debug("download cancelled");
             this.listener.onDownloadCancelled(this);
           } else {
-            if (this.total != expected) {
+            if (this.total < expected) {
```

Comment:
I prefer the outright paranoia of the original check, but I don't care much either way given the moribund nature of this bit of code. :wink:

Reply:
Yeah, this just prevents the code from throwing an error if the content length is zero, which is what should happen if we receive a gzip encoded response.